src/coreclr/jit/importer.cpp

```diff
@@ -7294,19 +7298,67 @@ void Compiler::impImportBlockCode(BasicBlock* block)
 
                     if (block->KindIs(BBJ_COND))
                     {
-                        if (op1->AsIntCon()->gtIconVal)
-                        {
-                            JITDUMP("\nThe conditional jump becomes an unconditional jump to " FMT_BB "\n",
-                                    block->GetTrueTarget()->bbNum);
-                            fgRemoveRefPred(block->GetFalseEdge());
-                            block->SetKindAndTargetEdge(BBJ_ALWAYS, block->GetTrueEdge());
-                        }
-                        else
+                        bool const      isCondTrue   = op1->AsIntCon()->gtIconVal != 0;
+                        FlowEdge* const removedEdge  = isCondTrue ? block->GetFalseEdge() : block->GetTrueEdge();
+                        FlowEdge* const retainedEdge = isCondTrue ? block->GetTrueEdge() : block->GetFalseEdge();
+
+                        JITDUMP("\nThe conditional jump becomes an unconditional jump to " FMT_BB "\n",
+                                retainedEdge->getDestinationBlock()->bbNum);
+
+                        fgRemoveRefPred(removedEdge);
+                        block->SetKindAndTargetEdge(BBJ_ALWAYS, retainedEdge);
+
+                        // If we removed an edge carrying profile, try to do a local repair.
+                        //
+                        if (block->hasProfileWeight())
                         {
-                            assert(block->NextIs(block->GetFalseTarget()));
-                            JITDUMP("\nThe block jumps to the next " FMT_BB "\n", block->Next()->bbNum);
-                            fgRemoveRefPred(block->GetTrueEdge());
-                            block->SetKindAndTargetEdge(BBJ_ALWAYS, block->GetFalseEdge());
+                            bool           repairWasComplete = true;
+                            weight_t const weight            = removedEdge->getLikelyWeight();
+
+                            if (weight > 0)
+                            {
+                                // Target block weight will increase.
+                                //
+                                BasicBlock* const target = block->GetTarget();
+                                assert(target->hasProfileWeight());
+                                target->setBBProfileWeight(target->bbWeight + weight);
```

Comment:
Maybe this is premature, but do you think we should consider changing the block weight API surface sooner rather than later? It might be useful as we do these profile fixups to have something like `incrementBBProfileWeight` that asserts the increment amount is positive, and clears the `BBF_RUN_RARELY` flag (until we decide to decouple the meaning of this flag from `BB_ZERO_WEIGHT`) -- and maybe doesn't touch the `BBF_PROF_WEIGHT` flag?

I'm happy to add some new helpers like this, if you think they'd be useful.

Reply:
Yes, we should do that -- want to get a bit more experience with the updates first, so we can see what patterns are common.